resources/PythonPip3Dsc/PythonPip3Dsc.psm1

```diff
@@ -205,4 +248,181 @@ function Invoke-Pip3
     }
 }
 
-#endregion Functions
\ No newline at end of file
+function TryGetRegistryValue{
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Key,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Property
+        )    
+
+    if (Test-Path -Path $Key)
+    {
+        try
+        {
+            return (Get-ItemProperty -Path $Key | Select-Object -ExpandProperty $Property)     
+        }
+        catch
+        {
+            Write-Verbose "Property `"$($Property)`" could not be found."
+        }
+    }
+    else
+    {
+        Write-Verbose "Registry key does not exist."
+    }
+}
+
+#endregion Functions
+
+$global:usePip3Exe = $false
+$global:pip3ExePath = Get-Pip3Path
+
+# Assert once that pip3 is already installed on the system.
+Assert-Pip3
+
+#region DSCResources
+<#
+.SYNOPSIS
+    The `Pip3Package` DSC Resource allows you to install, update, and uninstall Python packages using pip3.
+
+.PARAMETER SID
+    The security identifier. This is a key property and should not be set manually.
+
+.PARAMETER Exist
+    Indicates whether the package should exist. Defaults to $true.
+
+.PARAMETER Package
+    The name of the Python package to manage. This is a mandatory property.
+
+    For a list of Python packages, see https://pypi.org/.
+
+.PARAMETER Version
+    The version of the Python package to manage. If not specified, the latest version will be used.
+
+.PARAMETER Arguments
+    Additional arguments to pass to pip3.
+
+.PARAMETER InstalledPackages
+    A list of installed packages. This property is not configurable.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -ModuleName PythonPip3Dsc -Name Pip3Package -Method Set -Property @{ Package = 'flask' }
+
+    This example installs the Flask package.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -ModuleName PythonPip3Dsc -Name Pip3Package -Method Get -Property @{ Package = 'flask'; Version = '1.1.4' }
+
+    This example shows how to get the current state of the Flask package with version. If the version is not found, the latest version will be used if flask is found.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -ModuleName PythonPip3Dsc -Name Pip3Package -Method Get -Property @{ Package = 'django'; Exist = $false }
+
+    This example shows how Django can be removed from the system.
+#>
+[DSCResource()]
+class Pip3Package
+{
+    # DSCResource requires a key. Do not set.
+    [DscProperty(Key)]
+    [string]$SID
+
+    [DscProperty()]
+    [bool] $Exist = $true
+
+    [DscProperty(Mandatory)]
+    [string]$Package
+
+    [DscProperty()]
+    [string]$Version
+
+    [DscProperty()]
+    [string]$Arguments
+
+    [DscProperty(NotConfigurable)]
+    [hashtable[]]$InstalledPackages
+
+    [Pip3Package] Get()
+    {
+        $this.InstalledPackages = (GetInstalledPip3Packages | ConvertFrom-Json -AsHashtable)
+        $currentState = GetPip3CurrentState -Package $this.InstalledPackages -Parameters $this.ToHashTable()
```

Comment:
The parameters for this function are confusing to me. I initially thought Package was the package that we are trying to install and then realized that this was for the packages that are already installed. 

I think it would be better to simply iterate through the installed packages here and check if our desired package is in that list and returning that state. You might be able to get rid of that function entirely since it's not used anywhere else.